src/data/trainer-config.ts

```diff
@@ -1186,13 +1200,30 @@ function getSpeciesFilterRandomPartyMemberFunc(speciesFilter: PokemonSpeciesFilt
   };
 }
 
-function getRandomTeraModifiers(party: EnemyPokemon[], count: integer, types?: Type[]): PersistentModifier[] {
+/**
+ * Function to get random Tera modifiers for enemy trainers
+ * @param party {@linkcode EnemyPokemon} enemy trainer party
+ * @param count {@linkcode integer} how many Teras you want the trainer to have
+ * @param type {@linkcode Type} desired Tera type
+ * @param forceTeraFirstSlot {@linkcode boolean} Whether Trainer should attempt to Terastalize the mon in the first slot of signature species
+ * @returns {@linkcode PersistentModifier} an array of Tera modifiers for the enemy party
+ */
+function getRandomTeraModifiers(party: EnemyPokemon[], count: integer, type?: Type, forceTeraFirstSlot?: boolean): PersistentModifier[] {
   const ret: PersistentModifier[] = [];
   const partyMemberIndexes = new Array(party.length).fill(null).map((_, i) => i);
-  for (let t = 0; t < Math.min(count, party.length); t++) {
+  let t = 0; // Move this out of the for loop in case there's a forced tera slot
+  if (type && forceTeraFirstSlot) { // If there's a designated tera slot, essentially do an iteration of the for loop below with a set index and without checking count
```

Comment:
```suggestion
  if (!isNullOrUndefined(type) && forceTeraFirstSlot) { // If there's a designated tera slot, essentially do an iteration of the for loop below with a set index and without checking count
```
May require updating imports.